### PR TITLE
DDF-1772 Remove references to org.apache.servicemix.bundles.lucene

### PR DESCRIPTION
**Pull request by bdeining**
Remove references to org.apache.servicemix.bundles.lucene in offline-gazetteer and refactored spatial geocoding packages

@kcwire 
@dcruver is hero
@jrnorth 
@jaymcnallie

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/codice/ddf/pull/440" target="_blank">ReviewNinja

File: catalog/spatial/geocoding/spatial-geocoding-index/pom.xml [hunk 1]
```diff
@@ -78,10 +92,17 @@
                     <instructions>
                         <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
                         <Export-Package>org.codice.ddf.spatial.geocoding.index</Export-Package>
+                        <Private-Package>
+                            org.codice.ddf.spatial.geocoding.query,
```
Comment (stustison) [inline catalog/spatial/geocoding/spatial-geocoding-index/pom.xml]: Didn't you move everything from .query to .index, which would mean you don't need this private package here anymore?

File: catalog/spatial/geocoding/spatial-geocoding-index/pom.xml [hunk 2]
```diff
@@ -78,10 +92,17 @@
...
+                            org.apache.lucene.sandbox.queries
```
Comment (stustison) [inline catalog/spatial/geocoding/spatial-geocoding-index/pom.xml]: If you're embedding lucene-queries, you don't also need to private package one of the packages from that jar. You already have the entire jar embedded.

**Comment by bdeining** [comment]
@stustison Removed .query since it no longer exists.  The private package org.apache.lucene.sandbox.queries is part of lucene-sandbox, which is not embedded.

**Comment by stustison** [comment]
Is there a reason why you're private packaging that as opposed to just embedding it like the other lucene jars?

File: catalog/spatial/geocoding/spatial-geocoding-index/src/main/java/org/codice/ddf/spatial/geocoding/index/GeoNamesQueryLuceneDirectoryIndex.java [hunk 3]
```diff
@@ -12,7 +12,7 @@
  * <http://www.gnu.org/licenses/lgpl.html>.
  **/
 
-package org.codice.ddf.spatial.geocoding.query;
+package org.codice.ddf.spatial.geocoding.index;
```
Comment (kcwire) [inline catalog/spatial/geocoding/spatial-geocoding-index/src/main/java/org/codice/ddf/spatial/geocoding/index/GeoNamesQueryLuceneDirectoryIndex.java]: You could still leave these in the same package, just move them to the new project.

**Comment by bdeining** [comment]
@stustison The reason why it is not embedded is because org.apache.lucene.sandbox.queries.regex uses Jakarta Regexp, and would require that to be embedded as well.  Since org.apache.lucene.sandbox.queries.regex isn't used, this will clutter up the embedded dependencies for no real reason.  Additionally, as of 2011 Jakarta has been retired.

**Comment by stustison** [comment]
OK, sounds good. Private packaging should just be used sparingly, so I wanted to make sure that we checked all avenues.

**Comment by bdeining** [comment]
@keith Changed the PR to maintain the packages .query and .index. Also renamed the package to lucene.  @stustison this puts .query back in as a private package.

File: catalog/spatial/geocoding/spatial-geocoding-lucene/pom.xml [hunk 4]
```diff
@@ -78,10 +92,17 @@
                     <instructions>
                         <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
                         <Export-Package>org.codice.ddf.spatial.geocoding.index</Export-Package>
+                        <Private-Package>
+                            org.apache.lucene.sandbox.queries,
```
Comment (jrnorth) [inline catalog/spatial/geocoding/spatial-geocoding-lucene/pom.xml]: Might want to have a comment explaining why this is being private-packaged, based on the earlier discussion.

**Comment by bdeining** [comment]
@jrnorth Added a comment explaining why org.apache.lucene.sandbox.queries is private-packaged.

**Comment by jaymcnallie** [comment]
:+1:

**Pull request merged**